### PR TITLE
refactor: project url uses namespace and name instead of id

### PR DESCRIPTION
**Pull request by vfried**
This addresses #167 

Old URLs that where using the project id will be rerouted to the corresponding namespace/project-name url is possible.

There could be some edge cases that overlap with the new routing, priority will be given to the new schema, i think this edge cases don't exist in reality in renku.

Test Deployment:
https://virginiatest.dev.renku.ch/

Since this traverses all of the projects routing I would appreciate you testing it, if you find some error let me know :)

### CODE REVIEW

File: src/project/Project.js [hunk 1]
```diff
@@ -50,11 +50,29 @@ class View extends Component {
...
     }
+
+    // in case the route fails it tests weather it could be a projectid route
+    const routes=['overview','kus','ku_new','files','lineage','notebooks',
+      'data','workflows','settings','pending','launchNotebook','notebookServers'];
+    const available = this.props.core ? this.props.core.available : null;
+    const potentialProjectId = this.props.projectSlug.split('/')[0];
+    const potentialRoute = this.props.projectSlug.split('/')[1];
+
+    if(!available && !isNaN(potentialProjectId) && routes.indexOf(potentialRoute) > 0){
```
Comment (lorenzo-cavazzi) [inline src/project/Project.js]: I guess it's `>= 0` instead of `>`

**Comment by lorenzo-cavazzi** [comment]
I am trying to navigate on all pages.

I get an error when I try to visualize a notebook file while in "lineage view", like in this one
https://virginiatest.dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial/files/lineage/notebooks/zhbikes-notebook.ipynb

**Comment by lorenzo-cavazzi** [comment]
Another small issue. When I try to use an old project url, I get an error for the overview page.
E.G. https://virginiatest.dev.renku.ch/projects/95/ (with or without the final slash) doesn't work, but any sub url of that project works fine, like https://virginiatest.dev.renku.ch/projects/95/settings.

File: src/project/Project.js (hunk 1 above)
Comment (vfried) [reply src/project/Project.js]: Actually, there was a check missing, in case the projectSlug is undefined, meaning that is actually a projectId

Comment (vfried) [reply src/project/Project.js]: But well spotted anyways

**Comment by vfried** [comment]
> I am trying to navigate on different pages.
> 
> I get an error when I try to visualize a notebook file while in "lineage view", like in this one
> https://virginiatest.dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial/files/lineage/notebooks/zhbikes-notebook.ipynb

I just tried to navigate to the graph of a file with format .ipynb on de.renku.ch and it also fails. This is not a problem of the file path but i'm not sure where this problem comes from.

**Comment by vfried** [comment]
> E.G. https://virginiatest.dev.renku.ch/projects/95/ (with or without the final slash) doesn't work, but any sub url of that project works fine, like https://virginiatest.dev.renku.ch/projects/95/settings.

Fixed

**Comment by vfried** [comment]
Yes, sure... wouldn't it be feat(project)?

**Comment by lorenzo-cavazzi** [comment]
This has already been approved, but I found a couple of edge cases that may be confusing for the users

* Endless spinning wheel when trying to access a non existing project by id -- or a private project as well (https://virginiatest.dev.renku.ch/projects/9500/)
Screenshot from 2019-08-27 13-29-48

* Empty project when trying to access a wrong url path (https://virginiatest.dev.renku.ch/projects/abcdefg)
Screenshot from 2019-08-27 13-30-12

**Comment by vfried** [comment]
> This has already been approved, but I found a couple of edge cases that may be confusing for the users
> 
> * Endless spinning wheel when trying to access a non existing project by id -- or a private project as well (https://virginiatest.dev.renku.ch/projects/9500/)
>   Screenshot from 2019-08-27 13-29-48
> * Empty project when trying to access a wrong url path (https://virginiatest.dev.renku.ch/projects/abcdefg)
>   Screenshot from 2019-08-27 13-30-12

Ill fix this before we push it...

**Comment by vfried** [comment]
I fixed the not found pages, there where some errors and also some new pages added. I think they could have been part of a new issue but i wasn't sure, since is a small change i added it here. 

If you navigate to a path that doesn't exist inside a project you see this:
image

If you navigate to a path under /projects that doesn't exist you see this: 
image

If you navigate to a project with a numeric id that doesn't exist:
image

**Comment by rokroskar** [comment]
This doesn't seem like a huge problem but when I go to a URL for a blob directly, I see the 404 page flash for a split second... Then the correct page with the file content loads.

**Comment by vfried** [comment]
> This doesn't seem like a huge problem but when I go to a URL for a blob directly, I see the 404 page flash for a split second... Then the correct page with the file content loads.

I see what you mean... I'll try to fix it.

**Comment by lorenzo-cavazzi** [comment]
> I am trying to navigate on different pages.
> 
> I get an error when I try to visualize a notebook file while in "lineage view", like in this one
> https://virginiatest.dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial/files/lineage/notebooks/zhbikes-notebook.ipynb

info: this will be addressed by #582

**Comment by vfried** [comment]
> * We have 2 different "Project not found" pages. Wouldn't it be better to have a single one?

I added a new not found page but is for when the project was found but the url inside the project wasn't found... this allows the user to still navigate inside the project. If we put a simple project not found page the user can't navigate inside the project and it would also be technically wrong since the project is there.

**Comment by rokroskar** [comment]
Now that this is in place, we should discuss the larger topic of providing URIs for Renku entities. This should be in-sync with how entities are identified in the KG.
